update golang.org/x/oauth2 to support gcp workload identity

akhil-rane opened the pull request:
Workload identity federation will enable short lived credentials in a gcp
cluster. Current version golang.org/x/oauth2 does not support it.

ref: https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity
docs: https://github.com/openshift/cloud-credential-operator/pull/376

akhil-rane commented:
@JoelSpeed this looks good for a review. PTAL

JoelSpeed commented:
You'll need to work out why the goimports and generation is not working otherwise we won't be able to merge this

Is there an enhancement or relevant jira that you can point me to to understand the motivation behind this change?

akhil-rane commented:
> You'll need to work out why the goimports and generation is not working otherwise we won't be able to merge this
> 
> Is there an enhancement or relevant jira that you can point me to to understand the motivation behind this change?

I will work on fixing the goimports and generation but here is the jira for workload identity https://issues.redhat.com/browse/CCO-114

JoelSpeed commented:
> I will work on fixing the goimports and generation, here is the jira for workload identity https://issues.redhat.com/browse/CCO-114

We would normally expect some sort of enhancement to be created that allows us to understand the changes you are making to various components, at the moment the changes are fairly minor so I'm ok with not seeing one right now, but has one been written up? I'm aware this is similar to AWS STS and that required much larger changes, if this project will also require larger changes, I'd like to see an enhancement before we merge those

Please fixup the conflicts and I'll get this reviewed and labelled

akhil-rane commented:
> > I will work on fixing the goimports and generation, here is the jira for workload identity https://issues.redhat.com/browse/CCO-114
> 
> We would normally expect some sort of enhancement to be created that allows us to understand the changes you are making to various components, at the moment the changes are fairly minor so I'm ok with not seeing one right now, but has one been written up? I'm aware this is similar to AWS STS and that required much larger changes, if this project will also require larger changes, I'd like to see an enhancement before we merge those
> 
> Please fixup the conflicts and I'll get this reviewed and labelled

We do not have enhancement planned for this,  as you said nothing major is changing in ocp (mostly just bumping gcp sdk version and adding service account names to cred requests). But I have written detailed documentation that should give you a clear picture. Here is the PR for that https://github.com/openshift/cloud-credential-operator/pull/376

JoelSpeed commented:
Just had a read through the documentation, looks great thanks! I'm happy that it explains what's going on here sufficiently, we should add a link in the PR description to that doc if possible

JoelSpeed commented:
/lgtm

akhil-rane commented:
> Just had a read through the documentation, looks great thanks! I'm happy that it explains what's going on here sufficiently, we should add a link in the PR description to that doc if possible

I have fixed the conflicts and also added a doc link in the PR description.

JoelSpeed commented:
/override ci/prow/e2e-gcp-operator

This actually passed, the must gather at the end failed breaking the test

/test e2e-gcp

akhil-rane commented:
/test e2e-gcp

akhil-rane commented:
@JoelSpeed this PR has cleared the CI but needs approval.

JoelSpeed commented:
/approve
